Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -3857,4 +3935,126 @@ Vector<JS::NonnullGCPtr<Element>> Document::elements_from_point(double x, double
+
+    // the value of the name content attribute for all exposed embed, form, iframe, img, and exposed object elements
+    // that have a non-empty name content attribute and are in a document tree with document as their root;
+
+    // the value of the id content attribute for all exposed object elements that have a non-empty id content attribute
+    // and are in a document tree with document as their root; and
+
+    // the value of the id content attribute for all img elements that have both a non-empty id content attribute
+    // and a non-empty name content attribute, and are in a document tree with document as their root.
+
+    for (auto const& element : m_potentially_named_elements) {
+        if (!is_exposed(element))
+            continue;
+
+        if (is<HTML::HTMLObjectElement>(*element)) {
+            if (auto id = element->id(); id.has_value())
+                names.set(id.value());
+        }
+        if (is<HTML::HTMLImageElement>(*element)) {
+            auto maybe_name = element->name();
+            // Only set id if both name and id have value, for img elements. clear as mud
+            if (auto maybe_id = element->id(); maybe_name.has_value() && maybe_id.has_value())
+                names.set(maybe_id.value());
+        }
+
+        if (auto name = element->name(); name.has_value())
+            names.set(name.value());
+    }
+
+    Vector<FlyString> result;
+    result.ensure_capacity(names.size());
+    for (auto const& name : names)
+        result.unchecked_append(name);
+
+    return result;
+}
+
+// https://html.spec.whatwg.org/multipage/dom.html#dom-document-nameditem-filter
+static Vector<JS::NonnullGCPtr<DOM::Element>> named_elements_with_name(Document const& document, FlyString const& name)
+{
+    // Named elements with the name name, for the purposes of the above algorithm, are those that are either:
+
+    // - Exposed embed, form, iframe, img, or exposed object elements that have a name content attribute whose value is name, or
+    // - Exposed object elements that have an id content attribute whose value is name, or
+    // - img elements that have an id content attribute whose value is name, and that have a non-empty name content attribute present also.
+
+    Vector<JS::NonnullGCPtr<DOM::Element>> named_elements;
+    for (auto const& element : document.potentially_named_elements()) {
+        if (!is_exposed(*element))
+            continue;
+
+        if (is<HTML::HTMLObjectElement>(*element)) {
+            if (element->id() == name)
+                named_elements.append(element);
+        } else if (is<HTML::HTMLImageElement>(*element)) {
+            if (element->id() == name && element->name().has_value())
+                named_elements.append(element);
+        }
+
+        if (element->name() == name) {
+            named_elements.append(*element);
+        }
+    }
+
+    return named_elements;
+}
+
+// https://html.spec.whatwg.org/multipage/dom.html#dom-document-namedItem
```

Comment:
Same as the other PR this spec link doesn't work:
```
https://html.spec.whatwg.org/multipage/dom.html#dom-document-nameditem
```
(I didn't check the other links in this PR)

Reply:
d'oh. was just this one thankfully